lib/elixir/lib/map.ex

```diff
@@ -27,8 +30,48 @@ defmodule Map do
   @doc """
   Returns a new empty map.
   """
+  @spec new :: map
   def new, do: %{}
 
+  @doc """
+  Creates a `Map` from an enumerable.
+
+  Duplicated keys are removed, the latest one prevails.
+
+  ## Examples
+
+      iex> Map.new([{:b, 1}, {:a, 2}])
+      %{a: 2, b: 1}
+
+      iex> Map.new([a: 1, a: 2, a: 3])
+      %{a: 3}
+
+  """
+  @spec new(Enum.t) :: map
+  def new(enumerable) do
+    Enum.reduce(enumerable, %{}, fn {k, v}, acc -> put(acc, k, v) end)
+  end
+
+  @doc """
+  Creates a `Map` from an enumerable via the transformation function.
+
+  Duplicated entries are removed, the latest one prevails.
+
+  ## Examples
+
+      iex> Map.new([:a, :b], fn (x) -> {x, x} end)
```

Comment:
sorry jose, but if we are going to do that, why don't we do it now for this file only,
otherwise it will affect the history of this line unnecessarily